packager/VisibilityChecker.cc

```diff
@@ -496,31 +512,140 @@ class VisibilityCheckerPass final {
+                            }
+                        }
+
+                        for (auto &[package, packagesToReplace] : pass.convertTestImport) {
+                            for (auto pair : packagesToReplace) {
+                                replaceTestImportCapacitor.emplace_back(package, pair.first, pair.second);
+                            }
+                        }
                     }
                 }
             }
 
-            barrier.DecrementCount();
+            if (processedByThread > 0) {
+                toImportQ->push(std::move(importCapacitor), processedByThread);
+                replaceTestImportQ->push(std::move(replaceTestImportCapacitor), processedByThread);
+            }
         });
 
-        barrier.Wait();
+        UnorderedMap<core::packages::MangledName, PackageFixes> combinedFixes;
+        std::vector<std::tuple<core::packages::MangledName, core::packages::MangledName, bool>> importThreadResult;
+        for (auto result = toImportQ->wait_pop_timed(importThreadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer());
+             !result.done();
+             result = toImportQ->wait_pop_timed(importThreadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer())) {
+            if (!result.gotItem()) {
+                continue;
+            }
+            for (auto &[package, packageToImport, isTestImport] : importThreadResult) {
+                combinedFixes[package].toImport.emplace(packageToImport, isTestImport);
+            }
+        }
 
+        std::vector<std::tuple<core::packages::MangledName, core::packages::MangledName, core::Loc>>
+            replaceTestImportThreadResult;
+        for (auto result = replaceTestImportQ->wait_pop_timed(replaceTestImportThreadResult,
+                                                              WorkerPool::BLOCK_INTERVAL(), gs.tracer());
+             !result.done(); result = replaceTestImportQ->wait_pop_timed(replaceTestImportThreadResult,
+                                                                         WorkerPool::BLOCK_INTERVAL(), gs.tracer())) {
+            if (!result.gotItem()) {
+                continue;
+            }
+            for (auto &[package, packageToImport, loc] : replaceTestImportThreadResult) {
+                combinedFixes[package].convertTestImports.emplace(packageToImport, loc);
+            }
+        }
+
+        if (combinedFixes.empty()) {
+            return files;
+        }
+
+        for (auto &[pkg, fixes] : combinedFixes) {
+            auto &package = gs.packageDB().getPackageInfo(pkg);
+            auto newImportLoc = package.newImportLoc(gs, package);
+            fmt::memory_buffer autocorrect;
+            std::vector<core::AutocorrectSuggestion::Edit> edits;
+            auto fullPackageName = [&gs](const sorbet::core::packages::PackageInfo &pkg) {
+                return absl::StrJoin(pkg.fullName(), "::", core::packages::NameFormatter(gs));
+            };
```

Comment:
There's a couple problems here

- This allocates a lambda in a loop.
- This makes more people need to know about this weird `NameFormatter` thing.
- This code is the exact same as the `toString` method on `PackageName` in `packager.cc`, which we removed three calls to and have replace them with calls to this method.

Let's take that `toString` method, move it to `PackageInfo`, implement that method in `PackageInfoImpl`, and make a no-op version of it too. Then, VisibilityChecker won't need to add an import to `str_join.h`.